demo/index.ts

```diff
@@ -97,34 +98,53 @@ const {
   additionalRoleOptions,
 });
 
+const telemetryEventService = new UserTelemetryEventSender("example.com");
+
 const {
   plugin: elementPlugin,
   insertElement,
   nodeSpec,
   getElementDataFromNode,
-} = buildElementPlugin({
-  "demo-image-element": createDemoImageElement(onSelectImage, onDemoCropImage),
-  image: imageElement,
-  embed: createEmbedElement({
-    checkThirdPartyTracking: mockThirdPartyTracking,
-    convertTwitter: (src) => console.log(`Add Twitter embed with src: ${src}`),
-    convertYouTube: (src) => console.log(`Add youtube embed with src: ${src}`),
-    createCaptionPlugins,
-    targetingUrl: "https://targeting.code.dev-gutools.co.uk",
-  }),
-  interactive: createInteractiveElement({
-    checkThirdPartyTracking: mockThirdPartyTracking,
-    createCaptionPlugins,
-  }),
-  code: codeElement,
-  pullquote: pullquoteElement,
-  "rich-link": richlinkElement,
-  video: createVideoElement({
-    createCaptionPlugins,
-    checkThirdPartyTracking: mockThirdPartyTracking,
-  }),
-  membership: membershipElement,
-});
+} = buildElementPlugin(
+  {
+    "demo-image-element": createDemoImageElement(
+      onSelectImage,
+      onDemoCropImage
+    ),
+    image: imageElement,
+    embed: createEmbedElement({
+      checkThirdPartyTracking: mockThirdPartyTracking,
+      convertTwitter: (src) =>
+        console.log(`Add Twitter embed with src: ${src}`),
+      convertYouTube: (src) =>
+        console.log(`Add youtube embed with src: ${src}`),
+      createCaptionPlugins,
+      targetingUrl: "https://targeting.code.dev-gutools.co.uk",
+    }),
+    interactive: createInteractiveElement({
+      checkThirdPartyTracking: mockThirdPartyTracking,
+      createCaptionPlugins,
+    }),
+    code: codeElement,
+    pullquote: pullquoteElement,
+    "rich-link": richlinkElement,
+    video: createVideoElement({
+      createCaptionPlugins,
+      checkThirdPartyTracking: mockThirdPartyTracking,
+    }),
+    membership: membershipElement,
+  },
+  {
+    sendTelemetryEvent: (type: string) =>
+      telemetryEventService.addEvent({
+        app: "ProseMirrorElements",
+        stage: "TEST",
+        eventTime: new Date().toISOString(),
+        type,
+        value: true,
```

Comment:
Interesting – wonder if this is a hint that we should relax our interface, as sometimes value is not required?

Reply:
I had the same thought!